### PR TITLE
Add example for on-off simulation

### PR DESCRIPTION
**Pull request by AtreyeeS**
This addresses #4254

### CODE REVIEW

File: examples/tutorials/analysis-1d/spectral_analysis_rad_max.py [hunk 1]
```diff
@@ -349,4 +349,32 @@
...
+# A common way to check is a fit is biased is to simulate multiple datasets with
+# the obtained best fit model, and check the distribution of the fitted parameters.
+# Here, we show how to perform one such simulation
+#
+
+dataset_simulated = datasets.stack_reduce().copy(name="simulated_ds")
+simulated_model = best_fit_model.copy(name="simulated")
+dataset_simulated.models = simulated_model
+dataset_simulated.fake(
+    npred_background=dataset_simulated.counts_off * dataset_simulated.alpha
```
Comment (adonath) [inline examples/tutorials/analysis-1d/spectral_analysis_rad_max.py:366]: This line ignores the data from the in measurement. The predicted background counts in the Wstat case depend on the predict counts for the source. Instead they are given by https://docs.gammapy.org/1.0.1/api/gammapy.stats.get_wstat_mu_bkg.html#gammapy.stats.get_wstat_mu_bkg

However this is only valid for a given source model! We should at least mention this assumption...

**Review by QRemy — APPROVED** [review]
Thanks @AtreyeeS that's an interesting point to show in the doc.
In future we could have this in gammapy as a function that run a set of simulations+fit in parallel and return the distribution of fitted parameters. 
Maybe later we could also think of a tutorial dedicated to systematic studies based on simulations but also other methods like Jackknife masking part of the FoV or observations.

File: examples/tutorials/analysis-1d/spectral_analysis_rad_max.py [hunk 2]
```diff
@@ -349,4 +349,32 @@
 ax.set_ylim([1e-13, 1e-10])
 plt.show()
 
+
+######################################################################
+# Dataset simulations
+# -------------------
+#
+# A common way to check is a fit is biased is to simulate multiple datasets with
+# the obtained best fit model, and check the distribution of the fitted parameters.
```
[the remark this answers is not in the document]
Comment (AtreyeeS) [reply examples/tutorials/analysis-1d/spectral_analysis_rad_max.py:358]: Indeed that would be more useful. But we show this in the spectral_simulation notebook already, and I dont know how much of stuff we want to repeat..

**Review by QRemy — APPROVED** [review]

File: examples/tutorials/analysis-1d/spectral_analysis_rad_max.py [hunk 3]
```diff
@@ -349,4 +349,32 @@
...
 plt.show()
 
+
+######################################################################
+# Dataset simulations
+# -------------------
+#
+# A common way to check if a fit is biased is to simulate multiple datasets with
+# the obtained best fit model, and check the distribution of the fitted parameters.
+# Here, we show how to perform one such simulation
```
Comment (registerrier) [inline examples/tutorials/analysis-1d/spectral_analysis_rad_max.py:359]:
```suggestion
# Here, we show how to perform one such simulation assuming the measured off counts provide a good distribution of the background.
```

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS . This is good.

I just left a small suggestion to insist on the assumption that off counts provide a correct background distribution.

**Review by QRemy — APPROVED** [review]

**Pull request merged**